cookbooks/bcpc-hadoop/recipes/hdfsdu_pre_fetch.rb

```diff
@@ -0,0 +1,24 @@
+# Setup HDFSDU config
+
+set_hosts
+
+@rm_hosts = node[:bcpc][:hadoop][:rm_hosts]
+if defined?@rm_hosts and not @rm_hosts.empty? then
+   node.override[:hdfsdu][:jobtracker] = "#{float_host(@rm_hosts.first[:hostname])}:8032" 
+end
+
+node.override[:hdfsdu][:namenode] = "hdfs://#{node.chef_environment}"
+
+@oozie_hosts = node[:bcpc][:hadoop][:oozie_hosts]
+if defined?@oozie_hosts and not @oozie_hosts.empty? then
+   node.override[:hdfsdu][:oozie_url] = "http://#{float_host(@oozie_hosts.first[:hostname])}:11000/oozie"
+end
+
+node.override[:hdfsdu][:pig_jar] = '/usr/lib/pig/pig-0.12.0.2.0.11.0-1.jar'
+
+node.override[:hdfsdu][:oozie_frequency] = 120
+node.override[:hdfsdu][:oozie_timezone] = "EST"
+node.override[:hdfsdu][:oozie_start_time] = %x[date -u "+%Y-%m-%dT%H:00Z"].strip 
+# Setting the end date to 6 months from today?
+#node.override[:hdfsdu][:oozie_end_time] = %x[date -v+6m -u "+%Y-%m-%dT%H:00Z"].strip
+node.override[:hdfsdu][:oozie_end_time] = "2015-12-31T21:00Z"
```

Comment:
If there is a forever option in Oozie, that seems like the right answer here rather than an arbitary date (even if it's a dynamically long date).

Otherwise, if the job gets re-submitted when the coordinator expires, simply submitting the job for the current date plus a week could likely be useful too.